src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -20752,6 +20911,7 @@ void Compiler::fgDebugCheckBBlist(bool checkBBNum /* = false */, bool checkBBRef
 #endif // DEBUG
 
     fgDebugCheckBlockLinks();
+    fgFirstBBisScratch();
```

Comment:
Do we _always_ require the first BB to be scratch? Is that a new requirement?

Reply:
No.

That method is called here just as a check. It verifies that if there is a scratch bb, that other expected invariants hold.